woocommerce/woocommerce-gateway-stripe
# Source user from order when anonymous, on order-pay page (Fixes #1505)

Fixes #1505 

## Changes proposed in this Pull Request:

We noticed Stripe customers are duplicated in some scenarios.
Using the "Pay for Order" (order-pay) feature, we send the link to customers; the page can be accessed while anonymous.
When anonymous (only) on this page, the plugin can't find the user since it only checks `get_current_user_id()` and then it creates another Stripe customer.

Proposal is: on order-pay only, if user is anonymous only, get the user from the order (`$order->get_user_id()`). It will avoid duplicating customers in Stripe.

## Testing instructions to reproduce the bug and check the fix

1. Create a manual order in the backend
2. Select a customer that already paid with Stripe in the past, and add a product
3. Save the order
4. Send the link of the "Customer payment page" (looks like https://sitename/order/order-pay/ORDERID/?pay_for_order=true&key=wc_order_xxxxxxxx) to the customer
5. Customer accesses this page **anonymously** (not logged-in / private navigation)
6. Customer types a credit card in the Stripe Form and submit, payment is a success (Stripe test mode enabled)
7. In Stripe Dashboard (Payments page) you will see a new payment with a newly created customer called `Name: , Guest`. **With the fix, the customer will be the same as the user from the order. No new Customer is created**

---


**Post merge**
